src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Commit.cs

```diff
@@ -52,9 +52,18 @@ private CompletionProvider GetCompletionProvider(CompletionItem item)
             this.StopModelComputation();
 
             CompletionChange completionChange;
-            using (var transaction = new CaretPreservingEditTransaction(
+            using (var transaction = CaretPreservingEditTransaction.TryCreate(
                 EditorFeaturesResources.IntelliSense, TextView, _undoHistoryRegistry, _editorOperationsFactoryService))
             {
+                if (transaction == null)
+                {
+                    // This text buffer has no undo history and has probably been unmapped.
```

Comment:
@jasonmalinowski The reason this happens is that the ExpressionEditor Workspace actually closes the buffer: http://index/#Microsoft.VisualStudio.Activities/Microsoft/VisualStudio/Activities/ExpressionEditor/ExpressionEditorWindow.cs,314